AnnoDesigner/Models/IAdjecentCellGrouper.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections.Generic;
+using System.Windows;
+
+namespace AnnoDesigner.Models
+{
+    public class CellGroup<T>
```

Comment:
As far as I know the best practice in .Net is to put every class/interface in its own file.
So maybe move `CellGroup<T>` to its own file?
Also it is no private class, so one point more for its own file.